### PR TITLE
adding rmarinbe keymap for ergodash

### PR DESCRIPTION
**Pull request by rmarinbe**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c:2]: This is not needed, as it's set elsewhere, and included by default. 
```suggestion
```

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+};
+
+#ifdef AUDIO_ENABLE
+float tone_qwerty[][2]     = SONG(QWERTY_SOUND);
+#endif
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c:43]: Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+	for (uint16_t i = 0; i < 256; ++i) {
+		rgblight_sethsv_noeeprom( (i + user_hue) % 256, 255, 255);
+		wait_ms(5);
+	}
+	layer_state_set_user(layer_state);
+};
+
+
+// Turn on RGB underglow according to active layer
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c:62]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+	switch (biton32(state)) {
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c:63]:
```suggestion
	switch (get_highest_layer(state)) {
```

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+
+
+// Initialize rgblight
+void keyboard_post_init_user(void) {
+	rgblight_enable_noeeprom();
+	rgblight_mode_noeeprom(RGBLIGHT_MODE_STATIC_LIGHT);
+	layer_state_set_user(layer_state);
+	uint16_t user_hue = rgblight_get_hue();
+	for (uint16_t i = 0; i < 256; ++i) {
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c:53]: save you a bit of space/memory: 
```suggestion
	for (uint8_t i = 0; i <= 255; ++i) {
```

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+      rgblight_sethsv_noeeprom(100, 255, 255); 
+      break;
+	}
+	return state;
+};
+
+
+void matrix_scan_user(void) {
+    // Caps lock (blue)
+    if (host_keyboard_leds() & (1<<USB_LED_CAPS_LOCK)) {
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c:89]:
```suggestion
    if (host_keyboard_led_state().caps_lock) {
```

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+      backlight_enable();
+    }
+}
+
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+         print("mode just switched to qwerty and this is a huge string\n");
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c:101]:
```suggestion
```

File: keyboards/ergodash/rev1/keymaps/rmarinbe/rmarinbe.json [hunk 9]
```diff
@@ -0,0 +1,371 @@
+{
```
Comment (drashna) [inline keyboards/ergodash/rev1/keymaps/rmarinbe/rmarinbe.json:1]: This should be removed.

File: rmarinbe_keymap.json [hunk 10]
```diff
@@ -0,0 +1,371 @@
+{
```
Comment (drashna) [inline rmarinbe_keymap.json:1]: This file needs to be removed.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c (hunk 1 above)
Comment (rmarinbe) [reply keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c:16]: done

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c (hunk 2 above)
Comment (rmarinbe) [reply keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c]: done

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c (hunk 3 above)
Comment (rmarinbe) [reply keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c]: done

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c (hunk 4 above)
Comment (rmarinbe) [reply keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c]: done

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c (hunk 5 above)
Comment (rmarinbe) [reply keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c]: done

File: keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c (hunk 6 above)
Comment (rmarinbe) [reply keyboards/ergodash/rev1/keymaps/rmarinbe/keymap.c]: done

**Comment by rmarinbe** [comment]
closing pull request for now

**Pull request closed without merging**